### PR TITLE
Add missing dependency on Apache HTTPCore 4.3.3

### LINKED ISSUE #200: Possible missing dependency: Apache HttpCore 4.3.x
When upgrading our dependencies, we encountered this error in our stack trace:

```
Exception Thrown
org.elasticsearch.common.inject.CreationException: Guice creation errors:

1) Error injecting constructor, java.lang.NoClassDefFoundError: org/apache/http/util/Args
  at org.elasticsearch.discovery.ec2.Ec2Discovery.<init>(Unknown Source)
  while locating org.elasticsearch.discovery.ec2.Ec2Discovery
  while locating org.elasticsearch.discovery.Discovery
    for parameter 3 at org.elasticsearch.node.service.NodeService.<init>(Unknown Source)
  while locating org.elasticsearch.node.service.NodeService
Caused by: java.lang.NoClassDefFoundError: org/apache/http/util/Args
```

After checking through the pomfile for elasticsearch-cloud-aws, we didn't find any references to a specific version of Apache HTTPCore (so we weren't able to directly identify which version elasticsearch-cloud-aws depends on).

Should this dependency be added to the pomfile?

### PR DESCRIPTION
**Pull request by spencerwi**
Fixes #200

### CODE REVIEW

**Comment by dadoonet** [comment]
We don't explicitly use any HTTPCore class in this plugin so we should not declare any dependency to it.

If there is any concern with a dependency provided by a lib, then you need to explicitly define the desired lib in your own `pom.xml` and potentially exclude it from `cloud-aws` dependency or `Twilio` dependency.

TBH I think that Twilio should better change [its dependencies](https://github.com/twilio/twilio-java/blob/master/pom.xml#L42-51).

@tlrx WDYT?

**Comment by spencerwi** [comment]
This issue is not in cloud-aws but in ElasticSearch proper. Closing PR to get to the correct source of the problem.

**Comment by dadoonet** [comment]
In elasticsearch the dependency I can see is a test dependency.
You should not have an issue with that.
Other dependencies like this one are often shaded so you should not be concerned.
Did I miss something else?

File: pom.xml [hunk 1]
```diff
@@ -53,6 +53,12 @@
             <artifactId>elasticsearch</artifactId>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupid>org.apache.httpcomponents</groupid>
```
Comment (voho) [inline pom.xml:57]: Not sure if this is case sensitive, but it should be groupId with capital I.

**Comment by dadoonet** [comment]
This repository is not maintained anymore. But thanks for reporting.

**Pull request closed without merging**